galvan-support/src/main/java/org/terracotta/testing/rules/BasicExternalCluster.java

```diff
@@ -68,30 +84,31 @@
   private Thread shepherdingThread;
   private boolean isSafe;
 
-  BasicExternalCluster(File clusterDirectory, int stripeSize, List<File> serverJars, String namespaceFragment,
-                       String serviceFragment, int clientReconnectWindowTime, int failoverPriorityVoterCount,
-                       Properties tcProperties, Properties systemProperties1, String logConfigExt) {
-    if (clusterDirectory.exists()) {
-      if (clusterDirectory.isFile()) {
+  BasicExternalCluster(Path clusterDirectory, int stripeSize, Set<Path> serverJars, String namespaceFragment,
+                       String serviceFragment, int clientReconnectWindow, int voterCount, Properties tcProperties,
+                       Properties systemProperties, String logConfigExt, boolean convertConfig) {
```

Comment:
Pass in the startup builder here.  The legacy one lives here with galvan-support code.  The new dynamic startup builder lives with dynamic config in tc-platform

Reply:
Done. I used functions so that this core truly remains agnostic of the builders and convertors, and purely operates on data. Not proud of the ari-ness of the functions though. I can define interfaces in tc-core instead of defining these new functions. Let me know what you think.

Reply:
How about some kind of builder interface.

String[] startupArgs = new StartupBuilderImpl().kitLocation(Path).workingDirectory(Path).consistentStartup(boolean).build()

That way you can have some sane defaults in tc-core and tc-platform.  You can define the interface in galvan support.

Reply:
Had to rethink my approach for this. Good idea - cleaned up a lot of the mess I had created. Done!